### PR TITLE
add role advanced settings: token_expiry_mins and cert_expiry_mins

### PR DESCRIPTION
**Pull request by noy93845**

### CODE REVIEW

File: athenz/utils.go [hunk 1]
```diff
@@ -157,15 +181,36 @@ func flattenRoleMembers(list []*zms.RoleMember) []interface{} {
...
+			"review":     review,
 		}
 		roleMembers = append(roleMembers, member)
 	}
 	return roleMembers
 }
 
+func expandRoleSettings(settings map[string]interface{}) (int32, int32) {
+	return int32(settings["token_expiry_mins"].(int)), int32(settings["cert_expiry_mins"].(int))
+}
```
Comment (dvirguttman) [inline athenz/utils.go:197]: please get rid of this method. I think direct access to the settings array is more readable

File: athenz/resource_role.go [hunk 2]
```diff
@@ -65,7 +66,33 @@ func ResourceRole() *schema.Resource {
...
+							ValidateDiagFunc: validateDatePatternFunc(DATE_PATTERN, MEMBER_REVIEW_REMINDER),
+						},
+					},
+				},
+			},
+			"settings": {
+				Type:        schema.TypeSet,
+				Description: "Advanced settings",
+				Optional:    true,
+				MaxItems:    1,
```
Comment (dvirguttman) [inline athenz/resource_role.go:84]: why it is set of 1 element? 
can't it be just a simple map?

Comment (noy93845) [reply athenz/resource_role.go:84]: schema.TypeMap defines a dynamic map that has no defined keys...
If you want to use statically defined setting names this is the way to go.

File: docs/resources/athenz_role.md [hunk 3]
```diff
@@ -44,6 +44,10 @@ resource "athenz_role" "foo_role" {
     name = "domain3.user3"
     review = "2023-12-29 23:59:59"
   }
+  settings {
```
Comment (mendi160) [inline docs/resources/athenz_role.md:47]: It should be explained in the section "Argument Reference"

Comment (noy93845) [reply docs/resources/athenz_role.md:47]: added to the docs

**Review by mendi160 — APPROVED** [review]

**Review by dvirguttman — APPROVED** [review]

**Pull request merged**